source/proof_document/spec/proof_document_spec.rb

```diff
@@ -0,0 +1,233 @@
+            acuant_error:{ code:nil, message: nil}, billed: true, errors:{},
+            liveness_score: nil,
+            match_score: nil,
+            raw_alerts: [],
+            result: 'Passed',
+            success: true,
+          }
+        )
+
+        expect(a_request(:post, callback_url)).to_not have_been_made
+      end
+    end
+  end
+
+  describe '#proof' do
+    subject(:function) do
+      IdentityIdpFunctions::ProofDocument.new(event)
+    end
+
+    let(:document_proofer) { instance_double(IdentityDocAuth::Acuant::AcuantClient) }
+
+    before do
+      allow(function).to receive(:document_proofer).and_return(document_proofer)
+
+      stub_request(:post, callback_url).
+        with(headers: { 'X-API-AUTH-TOKEN' => idp_api_auth_token })
+    end
+
+    context 'with a successful response from the proofer' do
+      before do
+        expect(document_proofer).to receive(:post_images).
+          and_return(Proofer::Result.new)
+      end
+
+      it 'posts back to the callback url' do
+        function.proof
+
+        expect(WebMock).to have_requested(:post, callback_url)
+      end
+    end
+
+    context 'with an unsuccessful response from the proofer' do
+      before do
+        expect(document_proofer).to receive(:post_images).
+          and_return(Proofer::Result.new(exception: RuntimeError.new))
+      end
+
+      it 'posts back to the callback url' do
+        function.proof
+
+        expect(WebMock).to have_requested(:post, callback_url)
+      end
+    end
+
+    context 'with a connection error talking to the proofer' do
+      before do
+        allow(document_proofer).to receive(:post_images).
+          and_raise(Faraday::ConnectionFailed.new('error')).
+          and_raise(Faraday::ConnectionFailed.new('error')).
+          and_raise(Faraday::ConnectionFailed.new('error'))
+      end
+
+      it 'retries 3 times then errors' do
+        expect { function.proof }.to raise_error(Faraday::ConnectionFailed)
+
+        expect(WebMock).to_not have_requested(:post, callback_url)
+      end
+    end
+
+    context 'with a connection error posting to the callback url' do
+      before do
+        expect(document_proofer).to receive(:post_images).
+          and_return(Proofer::Result.new)
+
+        stub_request(:post, callback_url).
+          to_timeout.
+          to_timeout.
+          to_timeout
+      end
+
+      it 'retries 3 then errors' do
+        expect { function.proof }.to raise_error(Faraday::ConnectionFailed)
+
+        expect(a_request(:post, callback_url)).to have_been_made.times(3)
+      end
+    end
+
+    context 'when there are no params in the ENV' do
+      before do
+        ENV.clear
+
+        expect(document_proofer).to receive(:post_images).
+          and_return(Proofer::Result.new)
+      end
+
+      it 'loads secrets from SSM and puts them in the ENV' do
+        expect(function.ssm_helper).to receive(:load).with('document_proof_result_lambda_token').and_return(idp_api_auth_token)
```

Comment:
Turns out we don't have `_lambda_` in these secret values in SSM now, can we rename this to match? 66711bfcaf717ee7446f263570240275b3a1d205